### PR TITLE
Fix opertator e2e failures

### PR DESCRIPTION
**Pull request by irisdingbj**
It take time for operator to init watch and finish reconcile especially for the 1st time. So this makes operator e2e flaky.   add sometime to make sure istio operator finished his work before start the e2e test.

### CODE REVIEW

File: tests/e2e/framework/kubernetes.go [hunk 1]
```diff
@@ -1025,6 +1031,31 @@ EOF`, k.KubeConfig, dummyValidationRule)
 	return nil
 }
 
+func (k *KubeInfo) waitForIstioOperator() error {
+
+	get := fmt.Sprintf(`cat << EOF | kubectl --kubeconfig=%s get icp example-istiocontrolplane -n istio-operator
+-o yaml
```
Comment (howardjohn) [inline tests/e2e/framework/kubernetes.go]: really confused what this is doing?

Comment (irisdingbj) [reply tests/e2e/framework/kubernetes.go]: run `kubectl get icp example-istiocontrolplane -n istio-operator -o yaml` to make sure we have `Status` field updated for icp -- istio operator will set the status filed only if the reconcile fininshed.

Comment (howardjohn) [reply tests/e2e/framework/kubernetes.go]: but what do we have this EOF cat stuff?

File: tests/e2e/framework/kubernetes.go [hunk 2]
```diff
@@ -1025,6 +1031,30 @@ EOF`, k.KubeConfig, dummyValidationRule)
...
+EOF`, k.KubeConfig)
+	timeout := time.Now().Add(istioOperatorTimeout)
+	for {
+		if time.Now().After(timeout) {
+			return errors.New("timeout waiting for istio operator to deploy Istio")
+		}
+
+		out, err := util.ShellSilent(get)
+		if err == nil && strings.Contains(out, "Status") {
```
Comment (howardjohn) [inline tests/e2e/framework/kubernetes.go]: shouldn't we be checking for something more explicit like Status=success?

File: tests/e2e/framework/kubernetes.go [hunk 3]
```diff
@@ -90,6 +90,8 @@ const (
 	kubernetesReadinessTimeout        = time.Second * 180
 	kubernetesReadinessInterval       = 200 * time.Millisecond
 	validationWebhookReadinessTimeout = time.Minute
+	istioOperatorTimeout              = time.Second * 300
+	istioOperatorFreq                 = time.Minute
```
Comment (howardjohn) [inline tests/e2e/framework/kubernetes.go]: Seems way to long

File: tests/e2e/framework/kubernetes.go [hunk 4]
```diff
@@ -1025,6 +1031,30 @@ EOF`, k.KubeConfig, dummyValidationRule)
...
+		if time.Now().After(timeout) {
+			return errors.New("timeout waiting for istio operator to deploy Istio")
+		}
+
+		out, err := util.ShellSilent(get)
+		if err == nil && strings.Contains(out, "Status") {
+			break
+		}
+
+		log.Errorf("istio-operator is still deploying Istio: %v", err)
```
Comment (howardjohn) [inline tests/e2e/framework/kubernetes.go]: This is not an error, maybe warn?

File: tests/e2e/framework/kubernetes.go (hunk 2 above)
Comment (irisdingbj) [reply tests/e2e/framework/kubernetes.go]: The status filed is a struct and will require us to marshal/unmarshal to whole yaml to get the real status. So just make sure we contain Status here.

File: tests/e2e/framework/kubernetes.go (hunk 3 above)
Comment (irisdingbj) [reply tests/e2e/framework/kubernetes.go]: it wil take 2-3 minutes for the 1st time reconciling. So for sure we leave 5 minutes here.

**Comment by howardjohn** [comment]
5 min is ok but we should poll more frequently like 10s

File: tests/e2e/framework/kubernetes.go (hunk 4 above)
Comment (irisdingbj) [reply tests/e2e/framework/kubernetes.go]: fixed

File: tests/e2e/framework/kubernetes.go (hunk 1 above)
Comment (irisdingbj) [reply tests/e2e/framework/kubernetes.go]: removed EOF and cat stuff

**Comment by irisdingbj** [comment]
> 5 min is ok but we should poll more frequently like 10s
OK

**Pull request merged**